libs/core/langchain_core/tools.py

```diff
@@ -1318,3 +1352,17 @@ def invoke_wrapper(callbacks: Optional[Callbacks] = None, **kwargs: Any) -> Any:
             description=description,
             args_schema=args_schema,
         )
+
+
+def _get_runnable_config_param(func: Callable) -> Optional[str]:
+    if isinstance(func, functools.partial):
```

Comment:
Feel like we do this in a few places. Maybe something we should just make a util for?  (the get_type_hints specifically)

Reply:
maybe for a sep pr?